test/modules/core/lib/widget-manager.spec.js

```diff
@@ -5,8 +5,15 @@ import {WidgetManager} from '@deck.gl/core/lib/widget-manager';
 import {WebMercatorViewport} from '@deck.gl/core';
 
 class TestWidget {
-  constructor({id}) {
-    this.id = id;
+  constructor(props) {
+    this.id = props.id;
+    this.viewId = props.viewId || null;
+    this.placement = props.placement || 'top-left';
```

Comment:
I see, weather or not to include these members as props would be case-by-case. For instance, doesn't make sense for tooltip to have a placement prop.

Reply:
Not sure if the meaning of these constants can be extended, but tooltips could be positioned relative to the cursor.

Reply:
The placement for that use-case is "fill", meaning it can be anywhere in the map container. Marker would use the same placement, and also wouldn't expose placement.